tests/core/test_countries.py

```diff
@@ -1,137 +1,126 @@
-        simulation.calculate('non_existent_variable', 2013)
+@pytest.mark.parametrize("simulation", [({}, PERIOD)], indirect = True)
+def test_divide_option_on_month_defined_variable(simulation):
+    with pytest.raises(ValueError):
+        simulation.person("disposable_income", PERIOD, options = [populations.DIVIDE])
 
 
-def test_calculate_variable_with_wrong_definition_period(make_simulation):
-    simulation = make_simulation({})
+@pytest.mark.parametrize("simulation", [({}, PERIOD)], indirect = True)
+def test_divide_option_with_complex_period(simulation):
+    quarter = PERIOD.last_3_months
 
-    with raises(ValueError) as error:
-        simulation.calculate('basic_income', 2016)
+    with pytest.raises(ValueError) as error:
+        simulation.household("housing_tax", quarter, options = [populations.DIVIDE])
 
     error_message = str(error.value)
-    expected_words = ['period', '2016', 'month', 'basic_income', 'ADD']
+    expected_words = ["DIVIDE", "one-year", "one-month", "period"]
 
     for word in expected_words:
-        assert word in error_message, 'Expected "{}" in error message "{}"'.format(word, error_message)
-
+        assert word in error_message, f"Expected '{word}' in error message '{error_message}'"
 
-def test_divide_option_on_month_defined_variable(make_simulation):
-    simulation = make_simulation({})
-    with raises(ValueError):
-        simulation.person('disposable_income', "2016-01", options = [DIVIDE])
 
+def test_input_with_wrong_period(simulation_builder, tax_benefit_system):
+    year = str(PERIOD.this_year)
+    variables = {"basic_income": {year: 12000}}
+    simulation_builder.set_default_period(PERIOD)
 
-def test_divide_option_with_complex_period(make_simulation):
-    simulation = make_simulation({})
-    quarter = periods.period('2013-12').last_3_months
-    with raises(ValueError):
-        simulation.household('housing_tax', quarter, options = [DIVIDE])
+    with pytest.raises(ValueError):
+        simulation_builder.build_from_variables(tax_benefit_system, variables)
 
 
-def test_input_with_wrong_period(make_simulation):
-    with raises(ValueError):
-        make_simulation({'basic_income': {2015: 12000}})
+def test_variable_with_reference(make_simulation, isolated_tax_benefit_system):
+    variables = {"salary": 4000}
+    simulation = make_simulation(isolated_tax_benefit_system, variables, PERIOD)
 
+    result = simulation.calculate("disposable_income", PERIOD)
 
-def test_variable_with_reference(make_isolated_simulation):
-    tax_benefit_system = CountryTaxBenefitSystem()  # Work in isolation
-
-    simulation_base = make_isolated_simulation(tax_benefit_system, {'salary': 4000})
-
-    revenu_disponible_avant_reforme = simulation_base.calculate('disposable_income', "2016-01")
-    assert(revenu_disponible_avant_reforme > 0)
+    assert result > 0
 
     class disposable_income(Variable):
-        definition_period = MONTH
+        definition_period = periods.MONTH
```

Comment:
The import style for periods should (ie. `MONTH` vs `periods.MONTH`) should be consistent across the code base. It seems that most other tests in this PR ([this for example](https://github.com/openfisca/openfisca-core/blob/96a543320a26eaf66f971d563165e431bc727ba6/tests/core/test_formulas.py#L3)) import MONTH directly.

Reply:
Yes, I've tried to correct that here #992 :

- All classes are imported in an explicit way when it is possible to do so (cyclic dependencies)
- All helper functions and variables are imported with their parent module (as here for example)

I didn't want to clutter this pull request with that, so this is probably just an automatism and we could ignore it, but I'd expect all the code base to follow the `periods.MONTH` pattern.

Reply:
I will profit from this opportunity to add a first version of a style guideline. That will help, I hope, in subsequent pull requests to be building it progressively. I think this to add a lot of value as for OpenFisca "unwritten" style choices being largely undocumented.

Reply:
Changed.